scene/animation/animation_player.cpp

```diff
@@ -1072,25 +1072,31 @@ void AnimationPlayer::_animation_process2(double p_delta, bool p_started) {
 	accum_pass++;
 
 	bool seeked = c.seeked; // The animation may be changed during process, so it is safer that the state is changed before process.
+	float blend = 1.0; // First animation we play at 100% blend
+
 	if (p_delta != 0) {
 		c.seeked = false;
 	}
 
-	_animation_process_data(c.current, p_delta, 1.0f, seeked, p_started);
-
-	List<Blend>::Element *prev = nullptr;
-	for (List<Blend>::Element *E = c.blend.back(); E; E = prev) {
+	List<Blend>::Element *next = NULL;
+	for (List<Blend>::Element *E = c.blend.front(); E; E = next) {
 		Blend &b = E->get();
-		float blend = b.blend_left / b.blend_time;
+		_animation_process_data(b.data, p_delta, blend, false, false);
+		blend = 1.0 - b.blend_left / b.blend_time; // This is how much to blend the NEXT animation
```

Comment:
Isn't it `blend -= b.blend_left / b.blend_time;`? I don't know, but I still think the blend weight calculation is wrong.

To begin with, as I mentions in https://github.com/godotengine/godot/pull/70241#issuecomment-1356843003, it is wired that blend value could exceed 1.0. I think it is most straight forward to iterate once for the calculation of blend weights before all blend animations go into process, and after that iterate again for process.

Reply:
I've reversed the order in which the blends are processed so instead attempting to combine all the blends together like it was doing, I blend with the result of whatever blends are stacked up on top of each other.

For example, say anim1 is blending with anim2 and anim3 starts playing.

If we were 50% into the blend to anim2 and 25% into the blend of anim3, it would be like this:

anim1: blend of 1.0.  100% of animation is pose is used.
anim2: blend of 0.5.  50% of pose is used on top of everything before (in this case, just anim1)
anim3: blend of 0.25. 25% of the pose is used on top of everything before (the combined blend of anim 1 and 2).

The 1.0 - blend_left / blend_time is just to invert it to be fraction of the blend in instead of the blend out.  It's a little convoluted because the numbers/values were not meant to be processed in that order, but, fundamentally, I don't think the original logic could have worked.

Here, we're not getting 175% blending, as the values might imply, and we can't just weigh the blends before computing then combine them, because they need to blend *from the previous pose* not from the base pose, so we have to completely calculate the first anim, then blend with the combination of the first and second, then use that result to blend to the third and so on.  This should effectively allow unlimited blends.

Reply:
Well, I think it works decently better than before. I will approve as long as the above comment is corrected.